src/systems/system.C

```diff
@@ -288,7 +295,104 @@ void System::init_data ()
    this->init_matrices();
 }
 
+void System::reinit_mesh ()
+{
+  // Calling init() twice on the same system currently works evil
+  // magic, whether done directly or via EquationSystems::read()
+  //libmesh_assert(!this->is_initialized());
+
+  // First initialize any required data:
+  // either only the basic System data
+  if (_basic_system_only)
+    System::init_data();
+  // or all the derived class' data too
+  else
+    this->init_data();
```

Comment:
What exactly is going on here?  You init_data(), but then the rest of this function is mostly a copy of System::init_data()?  I had hoped we'd be refactoring to avoid code duplication here, not ending up with code duplication and adding *code path* duplication to boot!

Reply:
I forgot to remove the rest of the function after copying over from init and removing the out of place assert.